Fix default values.

yamadayy opened the pull request:
Fix default values. Names of some attributes and functions are fixed.

YoshiyukiYamada committed:
Fix default values.

xr0038 commented:
A test failed since the definition of `filter_efficiency` has changed.

```
_______________________________________ test_efficiency ________________________________________

    def test_efficiency():
        # number of mirror = 5, mirror reflection rate = 0.98, QE = 0.8, filter through put = 0.9 is assumed
        sg = Parameters.get_instance()
        # sg.set_number_of_mirrors(5)
        # sg.set_one_mirror_efficiency(0.98)
>       sg.set_filter_efficiency(0.95)
E       AttributeError: 'Parameters' object has no attribute 'set_filter_efficiency'

tests/mission/test_mission.py:28: AttributeError
```

YoshiyukiYamada committed:
Fix test code.

yamadayy commented:
失礼しました。test実行前にsetup.pyをするのを忘れていました。(setup.pyを使う環境になかなか慣れません)

YoshiyukiYamada committed:
Fix comment.

HajimeKawahara reviewed (approved):
Thanks a lot!

xr0038 commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -14,24 +15,38 @@ class Parameters:
+        pixel_size, detector_separation_x, detector_separation_y, n_col_ch,　n_row_ch, n_ch, n_ref_pix_pre,
+        n_ref_pix_post, (Detector specification: confidential)
+        maneuver_time, large_maneuver_time, (NEC report)
+        optics_efficiency, quantum_efficiency, filter_efficiency (Read tables made by Kataza-san)
```
`optics_efficiency`, `quantum_efficiency`, `filter_efficiency` の setter (`Efficiency` インスタンスを受け取って set するだけ) も必要だと思います．`short_wavelength_limit`, `long_wavelength_limit` との関係がちょっとあやしくなりますが，対応は先送りとしてよいかと……．

astronasutarou reviewed (changes requested):
今回の更新ポイントではないですが一箇所 critical な typo があったので修正をお願いします．

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -69,39 +84,42 @@ def __init__(self):
-        self.__npix_pre = 8
-        self.__npix_post = 8
-        self.__exposure_time = 10.0  # second(s)
+        self.__n_ref_pix_pre = 8
```
こうすると「検出器上下の ref. pix が 8 pix である」というのは分からなくはないのですが、サイドの ref. pix はいかがいたしましょうか…。n_ref_pix_[top|bottom|left|right] もありかもしれません。

yamadayy replied:
pre/postはそのまま、top/bottomを追加してみましたが、そのあたりは詳しくないので、具体的に「こうしましょう」という提案があった方がありがたいです。それと、detector_format_x/yとの関係はどうしたら良いでしょう。

tkamizuka replied:
その意味では n_ref_pix_[top|bottom|left|right] が提案のつもりでした。

前回の片ざさん風の考え方でいくと pre/post が top/bottom に相当するようにも思いましたが…考えてみるとどっち向きになるかは自明ではありませんね。けれどすでに x/y が導入されているので、普通に考えて -x 側を left、+x 側を right、-y 側を bottom、+y 側を top としてやはりn_ref_pix_[top|bottom|left|right] でいかがでしょうか。

tkamizuka replied:
> 私の質問があいまいで申し訳ありません。xとy、top/bottom/left/rightの問題は、読み出し方向と関連してどうしたら良いか、という質問でした。今のdetector_format_x/yの実装は、y側は123ピクセル×16チャンネル-8-8(今はpreとpostですが、これがtop/bottomなのかleft/rightなのか)、x側はn_row_ch = 1952としています。y側に細くx側に長い短冊がy方向にチャンネル数だけ並んでいます。y側がtop/bottomでx側は今は使ってませんがleft/rightで良いでしょうか。どうせ全部8なので数値としては変わらないのですが。(ここはまだ手を付けていません。)

現状頭にあった形状は [こちら](https://www-tap.scphys.kyoto-u.ac.jp/~yamada/wiki/data/?%E6%A4%9C%E5%87%BA%E5%99%A8%E9%96%A2%E9%80%A3%E3%81%AE%E6%83%85%E5%A0%B1#d8e5f438) の ROIC デザイン参考図のようなものでした。現状 imagesim としても同様に、x 方向 (横方向右向きのイメージ) に 123 px x 16 ch が並んでおり、y 方向 (縦方向上向きのイメージ) に 1968 px が並んでいる想定でおります。ただし読みだし方向が実はちょっと違っており、各 ch 左下の角から右に読んで、次は一行上の左から右に読んで…という想定をしております。参考図によれば横方向の読み出しが左半分と右半分で異なるようではありますが、そこはまだ対応できていません。

このイメージで、各 ch の col (縦方向・y 方向 の列) の数 `__n_col_ch` が 123 で、row (横方向・x 方向の列) の数 `__n_row_ch` が 1268 となります。左端・右端の ch では `__n_col_ch` が 123 というと、この中には 8 pix の ref. pix. が入る定義になっていますので、 `__n_row_ch` も ref. pix. を含めた 1268 px としておくのでいかがでしょうか。上下方向の読み出し方向は参考図に明記されていないので、その概念が不要な bottom/top としておけば問題ないと思いますが、imagesim に従うとすれば下から上と思って pre が bottom、post が top と思っていただいても良いのではないかと思います。

astronasutarou replied:
解決していましたら resolve をお願いいたしますー．

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -69,39 +84,42 @@ def __init__(self):
 # TODO: check it should be const or variable?
         self.__cell_pix = 13
         self.__use_M_flag = False
         self.__reference_wave_length = 1.4e-6
```
wave_length -> wavelength?

tkamizuka replied:
修正が見えていないように思いますが push をご確認いただけませんでしょうか (こちらのミスで見えていないようでしたら申し訳ありません)。

tkamizuka replied:
確認しました。ご対応いただきありがとうございます。

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -122,17 +140,21 @@ def aperture_inner_diameter(self):
-    def set_effective_focal_length(self, value):
+    def set_f_number(self, value):
         if self.__use_M_flag:
             return
```
やっぱり M はまだ使ったほうが良いのでしょうか…。

tkamizuka replied:
M の件は片ざさんに決めてもらうということで了解済みです。

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -181,53 +194,55 @@ def set_attitude_control_error_mas(self, value):
-    def set_one_mirror_efficiency(self, value):
-        self.__one_mirror_efficiency = value
+    def set_short_wavelength_limit(self, value):
+        f_name = "data/filter/filter" + str(int(value * 1e8)).zfill(3) + ".json"
```
フィルタのファイルの命名規則を縛ってしまう気がしますが問題ありませんでしょうか。

yamadayy replied:
Mが要るかどうかは片坐さんに判断してもらいましょう。大澤さんの提案で、efficeincyのインスタンスをセットするメソッドを作ったので、この命名規則で表せないものは直接インスタンスを与えれば解決できます。そこをparameters.pyで対応することはできますが、将来の課題ということでどうでしょう？

tkamizuka replied:
承知しました。

YoshiyukiYamada committed:
Fix type and apply comments by R. Ohsawa.

yamadayy commented:
Apply R. Ohsawa's comments.

tkamizuka reviewed (commented):
photoclass の lamb という attribute の名前が少しわかりにくい気がしましたが、とくに問題なければこのままでもよいと思います。あとは parameters.py についていくつかコメントをいたしました。何卒よろしくお願いいたします。

yamadayy commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -69,39 +84,42 @@ def __init__(self):
+        # detector temperature
+        spec_list = pkg_resources.resource_filename('telescope_baseline', 'data/qe/qe170.json')
+        self.__quantum_efficiency = Efficiency.from_json(spec_list)
+        # filter cut off frequency ???
```
直しました。

tkamizuka replied:
こちらも push をご確認いただけませんでしょうか。

tkamizuka replied:
outdated になったので別のコメントにうつります。

YoshiyukiYamada committed:
Apply comments by T. Kamizuka.

tkamizuka commented:
コメントを追記しました。修正項目の push のご確認と n_ref_pix 系の名前について、ご確認のほど何卒よろしくお願いいたしまsう。

yamadayy commented:
commit/pushができていなかった件、申し訳ありません。昨日、開発環境側での作業でcommit/pushしてないはずなのにcommit/pushできるファイルが無いと言ってきたので、すでにpushしたのかと勘違いしていました。再起動後、できるようになりましたので、とりあえず昨日までの変更はcommit/pushしておきました。見えてますでしょうか？

yamadayy commented:
私の質問があいまいで申し訳ありません。xとy、top/bottom/left/rightの問題は、読み出し方向と関連してどうしたら良いか、という質問でした。今のdetector_format_x/yの実装は、y側は123ピクセル×16チャンネル-8-8(今はpreとpostですが、これがtop/bottomなのかleft/rightなのか)、x側はn_row_ch = 1952としています。y側に細くx側に長い短冊がy方向にチャンネル数だけ並んでいます。y側がtop/bottomでx側は今は使ってませんがleft/rightで良いでしょうか。どうせ全部8なので数値としては変わらないのですが。(ここはまだ手を付けていません。)

xr0038 commented:
Files changed から _outdated_ なコメントをクリックすると過去のコードが見えるんですね……．

astronasutarou reviewed (approved):
更新ありがとうございました．

tkamizuka commented:
@yamadayy 更新いただきありがとうございます。commit 自体は見えました。
@xr0038 outdated 出ない状態で file changed の比較をするにはどうしたらよいのでしょう…。
やはり commit 見に行くしかないですか？

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -106,17 +106,19 @@ def __init__(self):
         spec_list = pkg_resources.resource_filename('telescope_baseline', 'data/qe/qe170.json')
         self.__quantum_efficiency = Efficiency.from_json(spec_list)
-        # filter cut off frequency ???
+        # filter cut on frequency ???
```
frequency -> wavelength でしょうか。細かくて申し訳ありませんがご確認のほど、よろしくお願いいたします。

yamadayy replied:
修正しました

tkamizuka replied:
ありがとうございます。

astronasutarou commented:
@tkamizuka いやー……どうすればいいんでしょうね…… commit で確認するのは良さそうですが，戻ってきて resolve を押すのは面倒ですね (unresolved conversation で merge を lock しているせいではあるんですが)．

YoshiyukiYamada committed:
frequency -> wavelength

yamadayy commented:
@tkamizuka しつこくてすみません。x/y問題ご確認お願いします。
(13:32のコメント)
私の質問があいまいで申し訳ありません。xとy、top/bottom/left/rightの問題は、読み出し方向と関連してどうしたら良いか、という質問でした。今のdetector_format_x/yの実装は、y側は123ピクセル×16チャンネル-8-8(今はpreとpostですが、これがtop/bottomなのかleft/rightなのか)、x側はn_row_ch = 1952としています。y側に細くx側に長い短冊がy方向にチャンネル数だけ並んでいます。y側がtop/bottomでx側は今は使ってませんがleft/rightで良いでしょうか。どうせ全部8なので数値としては変わらないのですが。(ここはまだ手を付けていません。)

tkamizuka commented:
x/y top/bottom/right/left 問題のコメントをいたしました。ご確認ください。 @yamadayy

yamadayy commented:
@tkamizuka 
「x/y top/bottom/right/left 問題のコメントをいたしました。」
私の再質問14:02以降のコメントは見えないのですが。11:19のもので答えになっているということでしょうか？

tkamizuka commented:
あれ、見えていませんか。確かによくわからない pending というのがついているからそのせいかもしれません。。

こちらにコピペします。

> 私の質問があいまいで申し訳ありません。xとy、top/bottom/left/rightの問題は、読み出し方向と関連してどうしたら良いか、という質問でした。今のdetector_format_x/yの実装は、y側は123ピクセル×16チャンネル-8-8(今はpreとpostですが、これがtop/bottomなのかleft/rightなのか)、x側はn_row_ch = 1952としています。y側に細くx側に長い短冊がy方向にチャンネル数だけ並んでいます。y側がtop/bottomでx側は今は使ってませんがleft/rightで良いでしょうか。どうせ全部8なので数値としては変わらないのですが。(ここはまだ手を付けていません。)

現状頭にあった形状は [こちら](https://www-tap.scphys.kyoto-u.ac.jp/~yamada/wiki/data/?%E6%A4%9C%E5%87%BA%E5%99%A8%E9%96%A2%E9%80%A3%E3%81%AE%E6%83%85%E5%A0%B1#d8e5f438) の ROIC デザイン参考図のようなものでした。現状 imagesim としても同様に、x 方向 (横方向右向きのイメージ) に 123 px x 16 ch が並んでおり、y 方向 (縦方向上向きのイメージ) に 1968 px が並んでいる想定でおります。ただし読みだし方向が実はちょっと違っており、各 ch 左下の角から右に読んで、次は一行上の左から右に読んで…という想定をしております。参考図によれば横方向の読み出しが左半分と右半分で異なるようではありますが、そこはまだ対応できていません。

このイメージで、各 ch の col (縦方向・y 方向 の列) の数 __n_col_ch が 123 で、row (横方向・x 方向の列) の数 __n_row_ch が 1268 となります。左端・右端の ch では __n_col_ch が 123 というと、この中には 8 pix の ref. pix. が入る定義になっていますので、 __n_row_ch も ref. pix. を含めた 1268 px としておくのでいかがでしょうか。上下方向の読み出し方向は参考図に明記されていないので、その概念が不要な bottom/top としておけば問題ないと思いますが、imagesim に従うとすれば下から上と思って pre が bottom、post が top と思っていただいても良いのではないかと思います。

tkamizuka reviewed (commented):
これを投げていないせいですかね。。

tkamizuka commented:
pending 消えました。github に不慣れで申し訳ありません。

YoshiyukiYamada committed:
x/y/pre/post problem is solved.

tkamizuka commented on src/telescope_baseline/tools/mission/parameters.py:
```diff
@@ -21,8 +21,8 @@ class Parameters:
-        pixel_size, detector_separation_x, detector_separation_y, n_col_ch,　n_row_ch, n_ch, n_ref_pix_pre,
-        n_ref_pix_post, (Detector specification: confidential)
+        pixel_size, detector_separation_x, detector_separation_y, n_col_ch,　n_row_ch, n_ch, n_ref_pix_right,
+        n_ref_pix_left, (Detector specification: confidential)
```
ここに `n_ref_pix_top`, `n_ref_pix_bottom` を加える必要はありますでしょうか。

yamadayy replied:
n_row_chを1952から1968にすると、detector_format_yを計算するのに必要になるので、残しておきます。

tkamizuka replied:
すみません。ちゃんと理解できていないのですが、コメントの趣旨は "n_ref_pix_right と n_ref_pix_left の記述があるのであれば n_ref_pix_top, n_ref_pix_bottom も記述しておかなくてもよいのでしょうか” という単純なものでした。変なことを申し上げているようでしたらご容赦ください。

astronasutarou replied:
こちら，コメント部分ですし特に異論なければ追記して完了としても問題ないでしょうか？

tkamizuka replied:
@yamadayy さんへのコメントということでよろしいでしょうか。@xr0038 
上塚としては問題ありません。

tkamizuka replied:
ご対応ありがとうございます @xr0038

astronasutarou replied:
`n_ref_pix_top`, `n_ref_pix_bottom` を追加しました．

tkamizuka reviewed (approved):
一つだけマイナーコメントをいたしました。 approve にしておきます。

yamadayy commented:
上塚さんのご指摘(x/y問題)を反映したつもりですので、ご確認下さい。

Commit:
modify the class document

Append `n_ref_pix_top` and `n_ref_pix_bottom` in the document of Parameters class.
